packages/wp-now/src/pool.ts

```diff
@@ -0,0 +1,180 @@
+import { NodePHP } from '@php-wasm/node';
+
+const Fatal = Symbol('Fatal');
+const Spawn = Symbol('Spawn');
+const Reap = Symbol('Reap');
+
+let childCount = 0;
+
+export class PoolInfo {
+	id = childCount++;
+	requests = 0;
+	started = Date.now();
+	active = false;
+}
+
+/**
+ * Maintains and refreshes a list of php instances
+ * such that each one will only be fed X number of requests
+ * before being discarded and replaced.
+ */
+export class Pool {
+	instances = new Map(); // php => PoolInfo
+
+	spawner: () => Promise<any>; // Callback to create new instances.
+	maxRequests: number; // Max requests to feed each instance
+	maxJobs: number; // Max number of instances to maintain at once.
+
+	notifiers = new Map(); // Inverted promises to notify async code of backlogged item processed.
+	running = new Set(); // Set of busy PHP instances.
+	backlog = []; // Set of request callbacks waiting to be run.
+
+	constructor({
+		spawner = async (): Promise<any> => {},
+		maxRequests = 2000,
+		maxJobs = 5,
+	} = {}) {
+		Object.assign(this, { spawner, maxRequests, maxJobs });
+		this[Reap]();
+		this[Spawn]();
```

Comment:
what's the purpose of using these symbols instead of simply using `reap`, `spawn`, and `fatal` as names?

Reply:
That's just a pattern I use for private methods.

Reply:
@dmsnell This has been refactored: https://github.com/WordPress/playground-tools/pull/110/files#diff-a7ba51ff344142e1e49c791cb452b0c5f9d941aaf83ce9ff11cc40ae01c21513R34